### PR TITLE
Checking for existing trusted_ca rather than creating an empty one.

### LINKED ISSUE #2188: Envoy fails to parse lds v1 responses
*Description*:

when running envoy version https://github.com/envoyproxy/envoy/commit/b62c541005b601e9fde8c2fa0399f2f6c935fc3c with v1 config and v1 lds envoy will fail to parse the lds response with the following provided in [this gist](https://gist.github.com/jvshahid/a4541bdccb954fd03bda7e1231d289e8)

*Repro steps*:

Start envoy with the following config:

```json
{
   "listeners" : [],
   "cluster_manager" : {
      "clusters" : [
         {
            "type" : "static",
            "connect_timeout_ms" : 250,
            "hosts" : [
               {
                  "url" : "tcp://127.0.0.1:8080"
               }
            ],
            "lb_type" : "round_robin",
            "name" : "0-service-cluster"
         },
         {
            "lb_type" : "round_robin",
            "hosts" : [
               {
                  "url" : "tcp://127.0.0.1:2222"
               }
            ],
            "connect_timeout_ms" : 250,
            "type" : "static",
            "name" : "1-service-cluster"
         },
         {
            "connect_timeout_ms" : 250,
            "hosts" : [
               {
                  "url" : "tcp://127.0.0.1:61003"
               }
            ],
            "lb_type" : "round_robin",
            "type" : "static",
            "name" : "lds-cluster"
         }
      ]
   },
   "admin" : {
      "address" : "tcp://127.0.0.1:9901",
      "access_log_path" : "/dev/null"
   },
   "lds" : {
      "refresh_delay_ms" : 1000,
      "cluster" : "lds-cluster"
   }
}
```

and the following listeners config returned by the lds server

```json
{
   "listeners" : [
      {
         "ssl_context" : {
            "cipher_suites" : "[ECDHE-RSA-AES256-GCM-SHA384|ECDHE-RSA-AES128-GCM-SHA256]",
            "cert_chain_file" : "/etc/cf-instance-credentials/instance.crt",
            "private_key_file" : "/etc/cf-instance-credentials/instance.key"
         },
         "address" : "tcp://0.0.0.0:61001",
         "name" : "listener-8080",
         "filters" : [
            {
               "config" : {
                  "stat_prefix" : "ingress_tcp-9534127d-306e-49b5-5158-9688cf1cd33b",
                  "route_config" : {
                     "routes" : [
                        {
                           "cluster" : "0-service-cluster"
                        }
                     ]
                  }
               },
               "type" : "read",
               "name" : "tcp_proxy"
            }
         ]
      },
      {
         "ssl_context" : {
            "private_key_file" : "/etc/cf-instance-credentials/instance.key",
            "cipher_suites" : "[ECDHE-RSA-AES256-GCM-SHA384|ECDHE-RSA-AES128-GCM-SHA256]",
            "cert_chain_file" : "/etc/cf-instance-credentials/instance.crt"
         },
         "address" : "tcp://0.0.0.0:61002",
         "name" : "listener-2222",
         "filters" : [
            {
               "name" : "tcp_proxy",
               "type" : "read",
               "config" : {
                  "stat_prefix" : "ingress_tcp-d784818b-5cc6-4335-5ad7-f3bec75347fd",
                  "route_config" : {
                     "routes" : [
                        {
                           "cluster" : "1-service-cluster"
                        }
                     ]
                  }
               }
            }
         ]
      }
   ]
}
```
watch envoy constantly failing to parse the lds output with the error given in [this gist](https://gist.github.com/jvshahid/a4541bdccb954fd03bda7e1231d289e8)

The issue seems to be in the translation from the lds v1 to v2. It looks like the trusted_ca filename is set to the empty string but instead should be left out completely.

### PR DESCRIPTION
**Pull request by alyssawilk**
*Description*:
Previously, we default added a trusted ca if one was not present, causing validation failures.

*Risk Level*: Low

*Testing*:
LDS repro test which would fail validation prior to the fix.

*Release Notes*:
n/a (minor bugfix)

Fixes  #2188

### CODE REVIEW

File: source/common/config/tls_context_json.cc [hunk 1]
```diff
@@ -40,8 +40,10 @@ void TlsContextJson::translateCommonTlsContext(
   translateTlsCertificate(json_tls_context, *common_tls_context.mutable_tls_certificates()->Add());
 
   auto* validation_context = common_tls_context.mutable_validation_context();
-  validation_context->mutable_trusted_ca()->set_filename(
-      json_tls_context.getString("ca_cert_file", ""));
+  if (validation_context->has_trusted_ca()) {
```
Comment (mattklein123) [inline source/common/config/tls_context_json.cc]: Shouldn't we be guarding this on the JSON object containing "ca_cert_file"? (Like "verify_certificate_hash")

File: test/server/lds_api_test.cc [hunk 2]
```diff
@@ -216,6 +217,79 @@ TEST_F(LdsApiTest, Basic) {
   EXPECT_EQ(18068408981723255507U, store_.gauge("listener_manager.lds.version").value());
 }
 
+TEST_F(LdsApiTest, Repro) {
```
Comment (mattklein123) [inline test/server/lds_api_test.cc]: nit: Better test name and overview comment?

**Comment by mattklein123** [comment]
Oh and regarding release notes: Historically a) we haven't release noted bug fixes and b) we have not been doing point releases (have been asking people to run master). I would suggest we don't do anything here and if there is big demand for point releases we can consider doing that and cherry picking.

File: source/common/config/tls_context_json.cc (hunk 1 above)
Comment (alyssawilk) [reply source/common/config/tls_context_json.cc]: We should indeed.  That's what I get for uploading on my way out the door :-P

plus side, no cavities?

File: test/server/lds_api_test.cc [hunk 3]
```diff
@@ -216,6 +217,81 @@ TEST_F(LdsApiTest, Basic) {
   EXPECT_EQ(18068408981723255507U, store_.gauge("listener_manager.lds.version").value());
 }
 
+// Regression test issue #2188 where an empty ca_cert_file field was created and caused and the LDS
```
Comment (mattklein123) [inline test/server/lds_api_test.cc]: typo: "and the"

**Comment by mattklein123** [comment]
Please merge master.

File: test/server/lds_api_test.cc [hunk 4]
```diff
@@ -216,6 +217,81 @@ TEST_F(LdsApiTest, Basic) {
...
+               }},
+               "type" : "read",
+               "name" : "tcp_proxy"
+            }}
+         ]
+      }}
+   ]
+  }}
+  )EOF";
+  std::string response2_json =
```
Comment (htuch) [inline test/server/lds_api_test.cc]: Nit: I would make these strings const.

**Pull request merged**